crates/astria-core/src/sequencer/v1alpha1/transaction/mod.rs

```diff
@@ -63,6 +64,17 @@ pub struct SignedTransaction {
 }
 
 impl SignedTransaction {
+    /// Returns the transaction hash.
+    ///
+    /// The transaction hash is calculated by protobuf-encoding the transaction
+    /// and hashing the resulting bytes with sha256.
+    #[must_use]
+    pub fn hash(&self) -> [u8; 32] {
```

Comment:
We can do this, but please change the name to something more verbose like `sha256_of_proto_encoding`.

The reason is that for an inherent method I expect a hash to a) be in terms of the rust memory representation (like if you `#[derive(Hash)]`), and b) be relatively cheap (just the cost of hashing, not an additional serialization + allocation).

Reply:
done!